pkg/wasm/cache.go

```diff
@@ -211,20 +254,43 @@ func (c *LocalFileCache) addEntry(key cacheKey, wasmModule []byte, f string) err
 	}
 
 	ce := cacheEntry{
-		modulePath: f,
-		last:       time.Now(),
+		modulePath:      f,
+		last:            time.Now(),
+		referencingURLs: sets.New(),
 	}
-	c.modules[key] = &ce
+	if needChecksumUpdate {
+		ce.referencingURLs.Insert(key.downloadURL)
+	}
+	c.modules[key.moduleKey] = &ce
 	wasmCacheEntries.Record(float64(len(c.modules)))
 	return nil
 }
 
 func (c *LocalFileCache) getEntry(key cacheKey) string {
 	modulePath := ""
 	cacheHit := false
+
+	// Only apply this for OCI image, not http/https because OCI image has ImagePullPolicy
+	// to control the pull policy, but http/https currently rely on existence of checksum.
```

Comment:
what about tagged oci url with sha256?

Reply:
In that case, we already have sha256, so the sha256 digest will be used for finding an entry in the cache. If there is no entry in cache, the tagged URL will be used for retrieving an image from registry server.

So, the case is working as is.